server/server.js

```diff
@@ -270,31 +269,41 @@ APP.post('/api/auth/logout', (req, res) => {
 // Handles any requests that don't match the ones above.
 // All other routing except paths defined above is done by React in the UI
 APP.get('*', async(req,res) =>{
-    var campaign, title, description, image;
+    var campaign, title, description, image, url;
     var splitURL = req.url.split('/'); 
     var campaignId = splitURL[splitURL.length -1]
     if(splitURL.length > 2) {
         try {
             const DB = CLIENT.db(DBNAME);
-            campaign = await DB.collection("campaigns").findOne({"_id" : campaignId});     
-            title = campaign.title.default;
-            description = campaign.description.default;
-            image = campaign.mainImageURL;   
+            campaign = await DB.collection("campaigns").findOne({"_id" : campaignId});
+                if(campaign){     
+                    title = campaign.title.default;
+                    description = campaign.description.default;
+                    image = campaign.mainImageURL;
+                    url = req.url;   
+                } else {
+                    title="This campaign is no longer available.";
+                    description="";
+                    image = "https://app.heo.finance/static/media/heo-logo.e772bc1b.png";
+                }
             } catch (err){
-                console.log(error);
+                console.log(err);
```

Comment:
the only problem with current logic is that if Mongo driver throws an exception we do not return any useful info in the metatags